soccerdata/sofascore.py

```diff
@@ -0,0 +1,265 @@
+                        'team': row['team']['name'],
+                        'MP': row['matches'],
+                        'W': row['wins'],
+                        'D': row['draws'],
+                        'L': row['losses'],
+                        'GF': row['scoresFor'],
+                        'GA': row['scoresAgainst'],
+                        'GD': row['scoresFor'] - row['scoresAgainst'],
+                        'Pts': row['points'],
+                    }
+                )
+            df = (
+                pd.DataFrame(mult_tables)
+                .set_index(idx)
+                .replace({'team': TEAMNAME_REPLACEMENTS})
+                .sort_index()[cols]
+            )
+        return df
+
+    def read_schedule(self) -> pd.DataFrame:
+        """Retrieve the game schedule for the selected leagues and seasons.
+
+        Returns
+        -------
+        pd.DataFrame
+        """
+        urlmask1 = SOFASCORE_API + "unique-tournament/{}/season/{}/rounds"
+        urlmask2 = SOFASCORE_API + "unique-tournament/{}/season/{}/events/round/{}"
+
+        cols = [
+            'round',
+            'week',
+            'date',
+            'home_team',
+            'away_team',
+            'home_score',
+            'away_score',
+            'game_id',
+        ]
+
+        df_seasons = self.read_seasons()
+        all_schedules = []
+        for (lkey, skey), season in df_seasons.iterrows():
+            # filepath = self.data_dir / filemask.format(lkey, skey)
+            url1 = urlmask1.format(season['league_id'], season['season_id'])
+            reader1 = self.get(url1)
+            season_data = json.load(reader1)
+            rounds = season_data['rounds']
+
+            for round in rounds:
+                url2 = urlmask2.format(season['league_id'], season['season_id'], round['round'])
+                reader2 = self.get(url2)
+                match_data = json.load(reader2)
+                for _match in match_data['events']:
+                    if _match['status']['code'] == 100 or _match['status']['code'] == 0:
+                        if _match['status']['code'] == 100:
+                            home_score = int(_match['homeScore']['current'])
+                            away_score = int(_match['awayScore']['current'])
+                        else:
+                            home_score = float('nan')  # type: ignore
+                            away_score = float('nan')  # type: ignore
+
+                        all_schedules.append(
+                            {
+                                'league': lkey,
+                                'season': skey,
+                                'round': "None",
+                                'week': _match['roundInfo']['round'],
+                                'date': datetime.datetime.fromtimestamp(
+                                    _match['startTimestamp']
+                                ).strftime("%Y-%m-%d %H:%M:%S%z"),
```

Comment:
Remove the strftime method here and drop the conversion to datetime on line 262.

Reply:
Modified as requested